src/k8s/pkg/k8sd/features/cilium/network_test.go

```diff
@@ -14,67 +17,128 @@ import (
 	"github.com/canonical/k8s/pkg/snap"
 	snapmock "github.com/canonical/k8s/pkg/snap/mock"
 	"github.com/canonical/k8s/pkg/utils"
-	"k8s.io/utils/ptr"
 )
 
 // NOTE(hue): status.Message is not checked sometimes to avoid unnecessary complexity
+type (
+	given struct {
+		config      types.Network
+		expectState helm.State
+		helmError   error
+	}
+	expect struct {
+		// tests for error
+		err []gtypes.GomegaMatcher
+		// tests for status
+		status []gtypes.GomegaMatcher
+		//	tests for helm
+		helm []gtypes.GomegaMatcher
+	}
+)
 
-func TestNetworkDisabled(t *testing.T) {
-	t.Run("HelmApplyFails", func(t *testing.T) {
-		g := NewWithT(t)
-
-		applyErr := errors.New("failed to apply")
-		helmM := &helmmock.Mock{
-			ApplyErr: applyErr,
-		}
-		snapM := &snapmock.Snap{
-			Mock: snapmock.Mock{
-				HelmClient: helmM,
+func TestNetwork(t *testing.T) {
+	helmErr := errors.New("failed to apply")
+	for _, tc := range []struct {
+		name   string
+		given  given
+		expect expect
+	}{
+		{
+			name: "NetworkDisabledHelmApplyFails",
+			given: given{
+				config:      types.Network{},
+				expectState: 0,
```

Comment:
maybe we can change this with `helm.Deleted` or the correct helm type for better readability.